wrappers/services.go

```diff
@@ -111,6 +111,15 @@ func StartServices(apps []*snap.AppInfo, inter interacter) (err error) {
 				return err
 			}
 
+			defer func(socketService string) {
+				if err == nil {
+					return
+				}
+				if e := sysd.Disable(socketService); e != nil {
+					inter.Notify(fmt.Sprintf("While trying to disable previously enabled socket service %q: %v", socketService, e))
+				}
+			}(socketService)
+
```

Comment:
why add a defer here, instead of adding it to the logic in the defer of the app, below?

Reply:
I figured the code would be easier to read when the cleanup is in direct proximity of the failed code. An altertive version is right here: https://github.com/bboozzoo/snapd/commit/b7794ed94ef903b8d30635252f76fafd705d2fca The drawback there is that it calls `systemctl stop <foo>.socket` even though (theoretically) there's no need to. If it feels cleaner, I can push it to this branch.

Reply:
I have a small preference for the altertive version :-) alternatively you could have a list of started sockets and loop over that in the defer, but that might be harder to follow
but it's a nit; you can land as is, or this, or with the list. My preferences are just that :-)

Reply:
Fair enough. Let's go with the alternative then :)